### ISSUE TITLE
Custom Favicons

### ISSUE THREAD

**Issue by ghost** [FreshRSS/FreshRSS#3789]
**Is your feature request related to a problem? Please describe.**
The possibility to assign custom favicons to feeds, to help separation.
The possibility to create sub-categories within the main categories. 

**Describe the solution you'd like**
A favicon upload option, for each feed on the Subscription Manager.  (and an option to not refresh the favicons)

**Describe alternatives you've considered**

**Comment by Alkarex** [comment]
We already have some issues discussing sub-categories, but let's keep this one for the custom favicons

**Comment by h4rp00n33r** [comment]
Hey folks, is there a manual way to edit or replace the used favicons?

**Comment by Alkarex** [comment]
> Hey folks, is there a manual way to edit or replace the used favicons?

Not at the moment, but PR welcome. We could both have a feed-level option, and an extension hook to allow FreshRSS extensions to control it.

For now, in theory, you could manually replace the favicons in `./FreshRSS/data/favicons/` but that is not so practical.

**Comment by h4rp00n33r** [comment]
> For now, in theory, you could manually replace the favicons in `./FreshRSS/data/favicons/` but that is not so practical.

Okay, thanks for your hint. Since there are only a few feeds that I want to customize, that's perfectly okay.
Is there anything else I need to consider after replacing the files?

**Comment by Alkarex** [comment]
> Is there anything else I need to consider after replacing the files?

Yes: the favicons are regularly refreshed so your changes risk being lost. So here are two manual options:

* Make the corresponding `.ico` file read-only (not tested, but should work)
* Make an HTTP redirection rule at your Web server level (e.g. Apache `Redirect permanent` or `RewriteRule`) pointing to whatever favicon you want

**Comment by h4rp00n33r** [comment]
> Yes: the favicons are regularly refreshed so your changes risk being lost. So here are two manual options:
> 
> * Make the corresponding `.ico` file read-only (not tested, but should work)
> * Make an HTTP redirection rule at your Web server level (e.g. Apache `Redirect permanent` or `RewriteRule`) pointing to whatever favicon you want

Okay, thx. I'll try the first option. :-)

**Comment by Inverle** [comment]
implemented in #7646

**Comment by Alkarex** [comment]
@h4rp00n33r If you are around, tests welcome of https://github.com/FreshRSS/FreshRSS/pull/7646

**Comment by h4rp00n33r** [comment]
> [@h4rp00n33r](https://github.com/h4rp00n33r) If you are around, tests welcome of [#7646](https://github.com/FreshRSS/FreshRSS/pull/7646)

Hey @Alkarex, that's great news. How can I test that feature? My installation is 1.26.4-dev.

![Image](https://github.com/user-attachments/assets/7d168724-a58a-4778-8e8f-5cb32fa79206)

**Comment by Inverle** [comment]
@h4rp00n33r 
via git (freshrss must be installed with git):
```bash
git fetch origin pull/7646/head:favicon-upload
git switch favicon-upload
```

**Comment by h4rp00n33r** [comment]
> [@h4rp00n33r](https://github.com/h4rp00n33r) via git (freshrss must be installed with git):
> 
> git fetch origin pull/7646/head:favicon-upload
> git switch favicon-upload

Thanks a lot. Unfortunately, I did not install FreshRSS via git. I'll wait until the release and give you feedback then. Okay?
I'm really looking forward to the new function. :-)

**Comment by Inverle** [comment]
@h4rp00n33r 

if it's a zip install then you can convert it to git like here: https://github.com/FreshRSS/FreshRSS/discussions/6769#discussioncomment-10531480 (but checkout `edge` instead of `latest` since you're on a dev version)

if it's a docker compose install then something like this: https://github.com/FreshRSS/FreshRSS/issues/7612#issuecomment-2907734376
^
```yaml
  freshrss:
    image: freshrss/freshrss:7646
    build:
      context: https://github.com/Inverle/FreshRSS.git#favicon-upload
      dockerfile: Docker/Dockerfile-Alpine
   ...
```

**Comment by Inverle** [comment]
also you can go to https://github.com/Inverle/FreshRSS/tree/favicon-upload and download the ZIP from there:

![Image](https://github.com/user-attachments/assets/30956856-b1ce-458b-aa42-888cfb12d50c)

**Comment by h4rp00n33r** [comment]
Okay, thx a lot. I will look into it. :-)
